sdk/typescript/package.json

```diff
@@ -5,31 +5,37 @@
 	"version": "0.36.0",
 	"license": "Apache-2.0",
 	"files": [
+		"builder",
 		"dist",
 		"src",
 		"CHANGELOG.md"
 	],
 	"engines": {
 		"node": ">=16"
 	},
-	"main": "./dist/index.js",
-	"module": "./dist/index.mjs",
-	"types": "./dist/index.d.ts",
+	"type": "commonjs",
+	"main": "./dist/cjs/index.js",
+	"module": "./dist/esm/index.js",
+	"types": "./dist/cjs/index.d.ts",
 	"exports": {
 		".": {
-			"types": "./dist/index.d.ts",
 			"source": "./src/index.ts",
-			"import": "./dist/index.mjs",
-			"require": "./dist/index.js"
+			"import": "./dist/esm/index.js",
+			"require": "./dist/cjs/index.js"
+		},
+		"./builder": {
+			"source": "./src/builder/index.ts",
+			"import": "./dist/esm/builder/index.js",
+			"require": "./dist/cjs/builder/index.js"
 		}
 	},
 	"scripts": {
 		"clean": "rm -rf tsconfig.tsbuildinfo ./dist",
 		"codegen:version": "node genversion.mjs",
 		"dev": "pnpm build:tsup --watch",
-		"build": "node genversion.mjs && pnpm build:types && pnpm build:tsup",
+		"build": "node genversion.mjs && pnpm build:package && pnpm build:tsup",
+		"build:package": "ts-node ../build-scripts/src/build-package.ts",
```

Comment:
Maybe worth making this package have a `bin` so that you can add the dep and reference it through the bin name instead of having the parent package need to call through a relative path . This also means the `ts-node` dep needs to live at the consuming package which is kind of weird.

Reply:
agreed, I had meant to do that, but kinda forgot after I got it working